Lib/fnmatch.py

```diff
@@ -135,18 +136,31 @@ def _translate(pat, star, question_mark):
                         if chunks[k-1][-1] > chunks[k][0]:
                             chunks[k-1] = chunks[k-1][:-1] + chunks[k][1:]
                             del chunks[k]
+
+                    if len(chunks)>1:
+                        char_range=set(range(ord(chunks[0][-1]), ord(chunks[-1][0])))
+
+                        question_mark_char=question_mark.replace('\\', '').replace('[', '').replace(']', '').replace('^', '')
+                        question_mark_char=set(map(ord, question_mark_char))
+
+                        if question_mark_char.intersection(char_range):
+                            inside_range=True
+
                     # Escape backslashes and hyphens for set difference (--).
                     # Hyphens that create ranges shouldn't be escaped.
                     stuff = '-'.join(s.replace('\\', r'\\').replace('-', r'\-')
                                      for s in chunks)
+
```

Comment:
please revert, we add blank lines to separate logical sections but this is still doing the same work